### ISSUE TITLE
MXF (playback) performance regression (on some clips)

### ISSUE THREAD

**Issue by manuelsongokuh** [MrKepzie/Natron#1095]
my video format is MXF (canon fx 105)
i input video readffmpeg and result VERY slow show..
see screen record:
http://sendvid.com/5tauzj6k

i test:
instead natron 1.2.1 is FASTER FASTER...
see screen record compare 1.2.1 vs 2.0rc5:
http://sendvid.com/isgo0uh7

**Comment by olear** [comment]
Natron2 is slower than Natron1 on (real-time) playback, but not as slow as your example (only able to view your first video), could you try remove cache/config? (rm -rf ~/.cache/INRIA/ ~/.config/INRIA/ ~/.local/share/INRIA/ , backup first! if you have important settings etc)

Btw, Natron1 didn't support MXF, at least not here.

**Comment by manuelsongokuh** [comment]
"could you try remove cache/config?" yes i done it, result same slow like as screen record..

natron1.2.1 works MXF, do you see my screen record? http://sendvid.com/isgo0uh7

**Comment by olear** [comment]
Tested various MXF, and they work ok in Natron2. If possible could you send me the MXF file your are testing with? (send me link(or the file if not too big) on email or as a pm on the forum)

**Comment by manuelsongokuh** [comment]
ok i will send you, for moment i's uploading.. ok? email is olear @  dracolinux  . org  right?

**Comment by olear** [comment]
Yes.

**Comment by manuelsongokuh** [comment]
i did sent email

**Comment by olear** [comment]
I'm able to reproduce with the clip you sent me, will do some more tests ....

**Comment by manuelsongokuh** [comment]
ok thanks you for more tests.. let me know as soon ok?

**Comment by manuelsongokuh** [comment]
what?! it's bug?!...ok...

**Comment by manuelsongokuh** [comment]
something news for this bug? because i'm work my office job.. but i dont force to you ok? i want to know only ok?

**Comment by olear** [comment]
Best you can do at the moment is to convert the video clip to an image sequence using ffmpeg (available with Natron, or use the one provided by your distro, you can also deinterlace the clip with ffmpeg).

**Comment by manuelsongokuh** [comment]
yes, this is sure for convert in sequence png with ffmpeg (also add de-interlaced) command and after import in natron but will be fast in viewer?

this issue is remain in bug of natron, because 1.2.1 is faster than 2.0 r5, i hope for future resolve this bug ok?
keep update for news of this bug ok?

**Comment by manuelsongokuh** [comment]
i will back office job, i will do new record with camcorder canon and short shot 5 or 2 minutes and send here footage of mxf for pubblic, so devs can download file footage mxf original and will does to test in natron,  i think will be good idea if site www.natron.fr new Subdomain example:
www.footage.natron.fr this is collection all little footage of test for devs will check and test like as laboratory test..?
good idea or is not helpful?

i will back office job to 7 or 11 January, i will send you file from my mega.nz ok?

**Comment by olear** [comment]
We know about the bug, so no need to make extra footage.

We currently host enough stuff related to Natron, hosting user uploads is not a priority, and not needed in my opinion since there are many upload options/sites online.

**Comment by manuelsongokuh** [comment]
ok. thank you

**Comment by MrKepzie** [comment]
Can you post a download link to that footage ?

**Comment by manuelsongokuh** [comment]
sorry me, i'm FEVER.. i wish to send you..i will back my job office to 11 January, monday i will send you footage, me sorry..
:-(

**Comment by olear** [comment]
I got the file (~700M) on my laptop, I will give it to Alex tomorrow.

**Comment by manuelsongokuh** [comment]
hello, i-m on bed, i see feedback here, i did told you alear, that video is not need to publicize, because that video is under rights of my office job, so I’m not authorized to public.
so i wish to send new footage from canon camcorder and rec shot in flower or fridge or objects, because i did send to olear footage video for "to confirm" ISSUE or BUG ok, video there is the face actress, that is not good for me public ok? not worry i will MUST rec for natron for best resolve problem, i wish to contribute Natron Team for best stable, of course.

problem is: i'm fever on bed, i cant go to fly (superman) to my office job and send you...joke :-)

i will wish new rec shot camcorder footage and send you public (not problem of legal rights of my office job ok?) because i wish to contribute all of you team of natron for true stable, i did done help you little bugs i was find. so i want again to find the bugs of natron for future stable ok?

good night!

**Comment by manuelsongokuh** [comment]
i'm thinking, tomorrow i will call my colleagues to make: rec to shot a flower or fridge and send me and i will re-send you ok?

**Comment by manuelsongokuh** [comment]
here link footage MXF original:
https://mega.nz/#!vpxUWToJ!cL4a3rtGfznHXJM-P03domBYLBKav3kQUoDuJ3QMVOE

FINALLY!!

**Comment by manuelsongokuh** [comment]
hi
i add here ohter link file MXF footage, scene: object with motion
https://mega.nz/#!D0ojBJQA!W476CDKWyBbDgQWucTaQt0WDBr8OmvJX9tT20Wdt8oM

**Comment by manuelsongokuh** [comment]
hi
natron 2.0 rc5, what is version of ffmpeg?
natron 1.2.1, what is version of ffmpeg?
i'm under investigation for this issue..

**Comment by manuelsongokuh** [comment]
hi, i'm back here..
i cant to use Natron for my mxf any..but maybe little fix (like as option) add bin ffmpeg version 2.7.x (for option) for default Natron 2.0 will be exist ffmpeg version 2.8.x but can add also ffmpeg version 2.7.x for who needed for old version of ffmpeg. because maybe i find the problem true is bug to ffmpeg but i'm not sure..

i did tested:
-kdenlive 0.9.10 using with ffmpeg 2.7.x for MXF worked as perfect, instead kdenlive 0.9.10 with ffmpeg 2.8.x not work high risk, instead kdenlive 15.12.1 (latest version) with ffmpeg 2.8.x with MXF is not worked but kdenlive latest has not option for ffmpeg 2.7.x so i'm back version old kdenlive 0.9.10 with ffmpeg 2.7.x worked as perfect.

-shotcut (mlt) latest version 16.2 with ffmpeg 2.7.x with MXF worked as perfect! shotcut has not option ffmpeg verison 2.8.x

**Comment by olear** [comment]
Will test against ffmpeg 2.7 .

**Comment by olear** [comment]
I didn't notice any big difference, but you can test for yourself:
```
cd <where_you_installed_Natron>/Natron2/Plugins/OFX/Natron/IO.ofx.bundle
cp -a Libraries Libraries.orig
cd Libraries
wget http://downloads.natron.fr/~olear/Natron/io-ffmpeg-2.7-Linux64.tgz
tar xvf io-ffmpeg-2.7-Linux64.tgz
```
Now run Natron.

**Comment by manuelsongokuh** [comment]
few minutes ago , test your tar, result nothing difference..same problem..maybe is problem from natron and not ffmpeg.. damn...i hope later will be fix..i do can patience for wait it ok?

**Comment by manuelsongokuh** [comment]
hi, i tested natron stable 2.0 with MXF result: nothing result, not fixed.. maybe next version..
but i wish ask you, can try compile ffmpeg version 3.0 for natron (development or snapshot)?

**Comment by olear** [comment]
That's why this issue is still open ;)

There where some changes in Read/Write-FFmpeg since v1, and sadly we have some performance issues on some clips. Changing FFmpeg version will not change this. We will look into the issue.

I recommend converting the video clip(s) to an image sequence, images will always be faster than videos, and it's easy and fast using ffmpeg on the commandline to convert the video.

**Comment by MrKepzie** [comment]
We found a fix, will be part of 2.0.1

**Comment by manuelsongokuh** [comment]
THANK YOU!
:+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1:
